src/main/kotlin/com/simplymerlin/minigameserver/minigame/oitc/phase/GamePhase.kt

```diff
@@ -0,0 +1,180 @@
+        ItemDropEvent::class.cancel(node)
+
+        node.addListener(PlayerRespawnEvent::class.java) {
+            spawn(it.player, false)
+            it.respawnPosition = randomRespawnLocation()
+        }
+
+        node.addListener(PlayerDeathEvent::class.java) {
+            it.chatMessage = null
+        }
+
+        node.addListener(FinalDamageEvent::class.java) {
+            val entity = it.entity
+            if (entity !is Player)
+                return@addListener
+            if (it.damageType.directEntity is Arrow) {
+                it.damage = 1000F
+            }
+            if (it.doesKillEntity()) {
+                sendDeathMessage(entity, it.damageType.entity as? Player, it.damageType)
+                val killer = it.damageType.entity
+                if (killer is Player && killer != entity) {
+                    onKill(killer)
+                }
+            }
+        }
+    }
+
+    override fun onUpdate() {
+
+    }
+
+    override fun onEnd() {
+        val winner = game.playerData.values.maxBy { it.score }.player
+        game.instance.showTitle(Title.title(winner.name.color(NamedTextColor.GREEN), Component.text("Has won.", NamedTextColor.GRAY)))
+        game.players.forEach {
+            sidebar.removeViewer(it)
+        }
+    }
+
+    private fun sendDeathMessage(victim: Player, killer: Player?, damageType: CustomDamageType) {
+        val skull = Component.text("☠", NamedTextColor.RED).append(Component.space())
+
+        val miniMessage: String
+
+        if (killer == null) {
+            miniMessage = if (damageType.isOutOfWorld) {
+                "<victim> fell out of the world."
+            } else if (damageType.isProjectile) {
+                "<victim> got shot."
+            } else {
+                "<victim> died."
+            }
+        } else {
+            if (victim == killer) {
+                miniMessage = if (damageType.isOutOfWorld) {
+                    "<victim> jumped into the void."
+                } else if (damageType.isProjectile) {
+                    "<victim> shot themselves."
+                } else {
+                    "<victim> killed themselves."
+                }
+            } else {
+                miniMessage = if (damageType.isOutOfWorld) {
+                    "<victim> got thrown in the void by <killer>."
+                } else if (damageType.isProjectile) {
+                    "<victim> got shot by <killer>."
+                } else {
+                    "<victim> got killed by <killer>."
+                }
+            }
+        }
+
+        val message = MiniMessage.miniMessage().deserialize(miniMessage,
+            Placeholder.component("victim", victim.name.color(NamedTextColor.GOLD)),
+            Placeholder.component("killer", killer?.name?.color(NamedTextColor.GOLD) ?: Component.empty()),
+            ).color(NamedTextColor.GRAY)
+
+        val component = Component.text()
+            .append(skull)
+            .append(Component.space())
```

Comment:
this space is added twice on line 85 as well

Reply:
nice catch!